wordpress-mobile/gutenberg-mobile
# Target gutenberg monorepo branch

## Description

The goal of this PR is to target the gutenberg monorepo branch. With this change `gutenberg` will receive more responsibility and gutenberg-mobile becomes a "customized experience" of the core RN editor. The idea is to have tests at both places. On gutenberg, integrated as part of the default CI and here as part of our usual contributions and releases.

Gutenberg PR: https://github.com/WordPress/gutenberg/pull/21332
WordPress-Android PR: https://github.com/wordpress-mobile/WordPress-Android/pull/11956
WordPress-iOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/14188

## Testing instructions

Check that all the following commands are working correctly. When running the "enhanced" gutenberg mobile you should see "Welcome to gutenberg for WP apps" as the post title while it should just display "Welcome to gutenberg" for the core version.

- [x] `npm run start:reset`
- [x] `npm run core android`
- [x] `npm run core ios`
- [x] `npm run genstrings`
- [x] `npm run bundle`
- [x] `npm run core test`
- [x] `npm run test:e2e:android:local`
- [x] `npm run test:e2e:ios:local`
- [x] `npm run core test:e2e:android:local`
- [x] `npm run core test:e2e:ios:local`
- [x] `npm run wpandroid`
- [x] Running the WPAndroid app in debug mode
- [x] Running the WPAndroid app in prod mode
- [x] Running the WPiOS app in debug mode
- [x] Running the WPiOS app in prod mode
- [x] CI tests running green in this PR
- [x] CI tests running green in the gutenberg PR

## Code review

The "Hide deleted files" option on github should help with the review: 
<img src="https://user-images.githubusercontent.com/230230/80514935-7b430e00-8981-11ea-8cb2-66c31519de59.png" width="300" />

PR submission checklist:
